### PR TITLE
Add sytem wide theming

### PR DESCRIPTION
**Pull request by WandersonAlves**
Hey, hello!

This PR tries to centralize the way the application handles colors. 

For `Screen` classes, the main idea is to use `Launcher.cpp` as the starter point for applying the color scheme. The code will grab the colors from the configuration and set then before `ScreenManager` gets called.

For other components like `ButtonMenu`, `ProgressBar` and `ScrollBar`, we simply get the colors from configuration and use them for configuring the components.

`SplashScreen` lives he own thing, after drawing the orca image, we clean the screen using the default values.

After discussing the PR, we can use this to update the documentation for dev usage, teaching how to create new `Screen`s, add new items on `MainMenuScreen`, adding new `ColorScheme`s and `Color`s etc.

As a starting question,

> There's more things that `Screen`s do to render their content (e.g `setTextDatum`, `setTextSize`, `setCursor`, etc). What of those things can be considered as defaults?

🚀

### CODE REVIEW

File: src/applications/launcher/Launcher.cpp [hunk 1]
```diff
@@ -29,8 +29,14 @@ Launcher *Launcher::getInstance()
 void Launcher::begin()
 {
     auto tft = DeviceBase::getInstance()->getInterfaces().displayInterface->getTFT();
-    auto settings = DeviceBase::getInstance()->getSettings();
+    auto device = DeviceBase::getInstance();
+    auto settings = device->getSettings();
     auto language = settings->getLanguage();
+    auto primaryColor = device->getSettings()->getPrimaryColor();
+    auto backgroundColor = device->getSettings()->getBackgroundColor();
```
Comment (marivaaldo) [inline src/applications/launcher/Launcher.cpp:36]: Please change to:

```
auto primaryColor = settings->getPrimaryColor();
auto backgroundColor = settings->getBackgroundColor();
```

Comment (marivaaldo) [inline src/applications/launcher/Launcher.cpp]: Please change to:

```cpp
auto primaryColor = settings->getPrimaryColor();
auto backgroundColor = settings->getBackgroundColor();
```

File: src/applications/services/gui/ProgressBar.cpp [hunk 2]
```diff
@@ -5,13 +5,20 @@ using namespace Applications::Services::GUI;
 ProgressBar::ProgressBar()
 {
     auto displayInterface = DeviceBase::getInstance()->getInterfaces().displayInterface;
-    auto displaySettings = displayInterface->getSettings();
     int progressBarMargin = 10;
-    this->_width = displaySettings.width - (progressBarMargin * 2);
+    this->_width = displayInterface->getSettings().width - (progressBarMargin * 2);
```
Comment (marivaaldo) [inline src/applications/services/gui/ProgressBar.cpp]: Please change to:

```cpp
this->_width = displaySettings.width - (progressBarMargin * 2);
```

**Review by marivaaldo — CHANGES_REQUESTED** [review]
Please see comments. Thank you so much!

**Comment by WandersonAlves** [comment]
Hey @marivaaldo, the issues found are fixed, please take a look again

**Review by marivaaldo — APPROVED** [review]

**Pull request merged**